klab.engine/src/main/java/org/integratedmodelling/klab/components/network/services/RoutingRelationshipInstantiator.java

```diff
@@ -133,12 +119,11 @@ public RoutingRelationshipInstantiator(IParameters<String> parameters, IContextu
 			this.geometryCollapser = GeometryCollapser
 					.valueOf(Utils.removePrefix(parameters.get("collapse_geometry", String.class)));
 		}
-		if (parameters.containsKey("server")) {
-			this.server = Server.valueOf(Utils.removePrefix(parameters.get("server", String.class)));
+		if (!parameters.containsKey("server")) {
```

Comment:
1. I always find it harder to reason about negative checks. Is there a away to express the same check without the not operator `!`? e.g. `parameters.get("server") == null || parameters.get("server").trim().isEmpty()` or something like that? 🤔 
I think we have `StringUtils.isEmpty()` too somewhere in the code, that checks for null or empty strings
2. Do we want to check if the URL is valid? at least well-formed? `new URL(server).toURI()` should raise an Exception if the URL is not valid

Reply:
Okay, I see why the negative check could be hard to read. I'll (try to) make it more legible.
Regarding the validity of the URL, I think that we could go a step further and check the status of the Valhalla server with single request.